packages/fx-core/src/core/error.ts

```diff
@@ -122,11 +122,20 @@ export function FunctionRouterError(func: Func) {
   );
 }
 
-export function ContextUpgradeError(error: any): FxError {
-  return new SystemError(
-    "ContextUpgradeError",
-    `Failed to update context: ${error.message}`,
-    CoreSource,
-    new Error().stack
-  );
+export function ContextUpgradeError(error: any, isUserError = false): FxError {
+  if (isUserError) {
+    return new UserError(
+      "ContextUpgradeError",
+      `Failed to update context: ${error.message}`,
+      CoreSource,
+      new Error().stack
```

Comment:
Why create a new Error to get stack, how about error.stack?

Reply:
Following the code structure above. Not quite sure about this. Maybe @jayzhang know more about this.

Reply:
OK

Reply:
This is just make the stack is not null for some cases when `error.stack` is `undefined`.
This can be improved by checking existence of `error.stack`.